lib/Dialect/FIRRTL/Transforms/LowerXMR.cpp

```diff
@@ -418,6 +420,40 @@ class LowerXMRPass : public LowerXMRBase<LowerXMRPass> {
     return success();
   }
 
+  LogicalResult handleTopModuleRefPorts() {
+    auto topModule = dyn_cast<FModuleOp>(*getOperation().getMainModule());
+    if (!topModule)
+      return success();
```

Comment:
I had to double check why this was necessary, but this is catching the case of an external module being the top module.

Reply:
Is this even allowed?  Verify seems to say allowed.  Spec implies there has to be a real module, but is ambiguous.

Reply:
Both SFC and MFC accept the top module being an external module. We could disallow it without consequence I think.

Right now you can put an external module as the top module in a circuit and use that as a separately compiled unit. However, the compilation of that is basically a no-op. The client using this also has to have an external module pointing at this circuit containing another extmodule. I don't see why we need to enable such redirection (duplication?).